pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongHashMap.java

```diff
@@ -333,25 +334,23 @@ V get(long key, int keyHash) {
                         if (!acquiredLock) {
                             stamp = readLock();
                             acquiredLock = true;
+
+                            // update local variable
+                            keys = this.keys;
+                            values = this.values;
+                            bucket = signSafeMod(keyHash, values.length);
                             storedKey = keys[bucket];
                             storedValue = values[bucket];
                         }
 
-                        if (capacity != this.capacity) {
-                            // There has been a rehashing. We need to restart the search
-                            bucket = keyHash;
-                            continue;
-                        }
-
                         if (storedKey == key) {
                             return storedValue != DeletedValue ? storedValue : null;
                         } else if (storedValue == EmptyValue) {
                             // Not found
                             return null;
                         }
                     }
-
-                    ++bucket;
+                    bucket = (bucket + 1) & (values.length - 1);
```

Comment:
just wondering if `signSafeMod` should be used here:
```suggestion
                    bucket = signSafeMod(bucket + 1, values.length);
```
Please check this also for other locations.

Reply:
We iterate the array here. As the length of array is an exponential power of 2, `&(values.length - 1)` is equal to  `mod values.length`.

Reply:
> As the length of array is an exponential power of 2, `&(values.length - 1)` is equal to `mod values.length`.

Oh, that's the trick that is played here...